libs/wingc/src/lsp/sync.rs

```diff
@@ -120,7 +120,7 @@ pub fn on_document_did_change(params: DidChangeTextDocumentParams) {
 fn partial_compile(
 	source_path: &Path,
 	source_text: String,
-	mut types: &mut Types, // TODO: does this need to be shared between recompiles?
+	mut types: &mut Types,
```

Comment:
It's important to still keep `Types` around as long as possible in-between compilations because we need to access raw pointers to data it contains. But once we're about to compile again we know that we won't need the data inside it anymore.